codex-process-data-transfer/src/main/java/de/netzwerk_universitaetsmedizin/codex/processes/data_transfer/service/send/ResolvePsn.java

```diff
@@ -107,12 +126,18 @@ private String resolvePseudonymAndUpdatePatient(Patient patient)
 		return pseudonym;
 	}
 
-	private String getBloomFilter(Patient patient)
+	private Optional<String> getBloomFilter(Patient patient)
 	{
 		return patient.getIdentifier().stream().filter(Identifier::hasSystem)
 				.filter(i -> NAMING_SYSTEM_NUM_CODEX_BLOOM_FILTER.equals(i.getSystem())).filter(Identifier::hasValue)
-				.findFirst().map(Identifier::getValue).orElseThrow(() -> new RuntimeException(
-						"No bloom filter present in patient " + patient.getIdElement().getValue()));
+				.findFirst().map(Identifier::getValue);
+	}
+
+	private Optional<String> getLocalPseudonym(Patient patient)
+	{
+		return patient.getIdentifier().stream().filter(Identifier::hasSystem)
+				.filter(i -> RFC_4122_SYSTEM.equals(i.getSystem())).filter(Identifier::hasValue).findFirst()
```

Comment:
If the usage of `urn:ietf:rfc:4122` for local pseudonyms and with that the semantics of RFC4122 are not an external requirement, we may want to define a NamingSystem similar to http://www.netzwerk-universitaetsmedizin.de/sid/bloom-filter, ...

As we are using https://www.medizininformatik-initiative.de/fhir/core/modul-person/StructureDefinition/PatientPseudonymisiert as the required profile for Patient resources, we may want to find the pseudonym identifier via the pattern defined in the profile:
```
{
  "type": {
    "coding": [
      {
        "system": "http://terminology.hl7.org/CodeSystem/v3-ObservationValue",
        "code": "PSEUDED"
      }
    ]
  }
}
```

Reply:
Currently, they are an external requirement, but I think you are correct and we should allow search by type. I would suggest to implement this in an upcoming release.

Reply:
concur, please create GitHub Issue.

Reply:
see ##125